src/swarm-lang/Swarm/Language/Context.hs

```diff
@@ -325,21 +320,24 @@ buildCtxMap m (CtxTree h s) = do
       CtxDelete x t s1 -> buildCtxMap (M.insert x t m) s1
       CtxUnion s1 s2 -> buildCtxMap m s1 *> buildCtxMap m s2
 
+newtype ConstHash v t = ConstHash CtxHash
+  deriving (Eq, Show, Generic, ToJSON, FromJSON)
```

Comment:
So this is a double const at the type level?

Reply:
Yes, it is.  At first I tried just using `Const (Const CtxHash)` but that didn't work for some reason.  I don't remember why.